core/domain/exp_jobs_one_off.py

```diff
@@ -831,3 +831,63 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class EmptySubtitledHtmlAuditJob(jobs.BaseMapReduceOneOffJobManager):
+    """One off job to audit explorations having empty subtitled html and outputs
+    the content_id of such object.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_services.get_exploration_from_model(item)
+        exp_rights = rights_manager.get_exploration_rights(item.id)
+
+        state_name_to_content_id_list = {}
+        for state_name, state in exploration.states.iteritems():
+            empty_html_content_id_list = []
+            content = state.content
+            if not (content.html).strip():
+                empty_html_content_id_list.append(content.content_id)
+
+            for answer_group in state.interaction.answer_groups:
+                feedback = answer_group.outcome.feedback
+                if not feedback.html.strip():
+                    empty_html_content_id_list.append(feedback.content_id)
+
+            default_outcome = state.interaction.default_outcome
+            if default_outcome and not default_outcome.feedback.html:
+                empty_html_content_id_list.append(
+                    default_outcome.feedback.content_id)
+
+            for hint in state.interaction.hints:
+                hint_content = hint.hint_content
+                if not hint_content.html:
+                    empty_html_content_id_list.append(hint_content.content_id)
+
+            solution = state.interaction.solution
+            if solution and not solution.explanation.html:
```

Comment:
Should this be `if solution is not None`?

Reply:
Done!